### PR TITLE
adds docker related files and configuration updates

### LINKED ISSUE #324: Create a docker script
Create a docker script to allow others to try out Nurax in a low barrier way.

### PR DESCRIPTION
**Pull request by revgum**
Fixes #324

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
Thanks so much for doing this @revgum, it ran without problems for me. I've made a couple of little suggestions on the DOCKER.md

**Review by ghost — APPROVED** [review]

**Review by phuongdh — COMMENTED** [review]
Good stuff. I have one small comment about file ownership. Rails runs under `root` user in the container so every file it creates in the bind-mounted volume is owned by `root`, including `fits.log`, uploads and derivatives. It's not a deal-breaker but what do you think about adding some instruction on how to run `nurax` as the host's uid/gid?

File: build/create_solr_index.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+set -e
+
+SOLR_HOST=$(echo $SOLR_URL | cut -d '/' -f 1,2,3)
+
+# Upload the existing hyrax SOLR configurations from solr/config.zip
```
Comment (phuongdh) [inline build/create_solr_index.sh]: There is a Solr trick that I've used for Avalon, mounting the `solr/config` folder and precreating core on startup (nothing happens if core already exists though). This way we won't have to maintain a separate, potentially out-of-sync `solr/config.zip` and simplify our setup.
```
  solr:
    image: solr:6.6.4-alpine
    volumes:
      - ./solr/config:/opt/solr/avalon_conf
      - solr:/opt/solr/server/solr/mycores
    entrypoint:
      - docker-entrypoint.sh
      - solr-precreate
      - avalon
      - /opt/solr/avalon_conf
```

**Review by phuongdh — APPROVED** [review]
Looking good!

**Pull request merged**